OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameCashCounterLogic.cs

```diff
@@ -9,22 +9,68 @@
  */
 #endregion
 
+using System;
+using System.Linq;
+using OpenRA.Mods.Common.Traits;
 using OpenRA.Traits;
 using OpenRA.Widgets;
 
 namespace OpenRA.Mods.Common.Widgets.Logic
 {
 	public class IngameCashCounterLogic : ChromeLogic
 	{
+		const float DisplayFracPerFrame = .07f;
+		const int DisplayDeltaPerFrame = 37;
+
+		readonly World world;
+		readonly Player player;
+		readonly PlayerResources playerResources;
+
+		int nextCashTickTime = 0;
+		int displayResources;
+
 		[ObjectCreator.UseCtor]
 		public IngameCashCounterLogic(Widget widget, World world)
 		{
-			var playerResources = world.LocalPlayer.PlayerActor.Trait<PlayerResources>();
+			this.world = world;
+			player = world.LocalPlayer;
+			playerResources = player.PlayerActor.Trait<PlayerResources>();
+			displayResources = playerResources.Cash + playerResources.Resources;
+
 			var cash = widget.Get<LabelWithTooltipWidget>("CASH");
 			var label = cash.Text;
 
-			cash.GetText = () => label.F(playerResources.DisplayCash + playerResources.DisplayResources);
+			cash.GetText = () => label.F(displayResources);
 			cash.GetTooltipText = () => "Silo Usage: {0}/{1}".F(playerResources.Resources, playerResources.ResourceCapacity);
 		}
+
+		public override void Tick()
+		{
+			if (nextCashTickTime > 0)
+				nextCashTickTime--;
+
+			var actual = playerResources.Cash + playerResources.Resources;
+
+			var diff = Math.Abs(actual - displayResources);
+			var move = Math.Min(Math.Max((int)(diff * DisplayFracPerFrame), DisplayDeltaPerFrame), diff);
+
+			if (displayResources < actual)
+			{
+				displayResources += move;
```

Comment:
You can implement a minor perf / memory churn improvement here by caching `displayLabel = label.F(displayResources)` and binding `cash.GetText = () => displayLabel` in the constructor.